librz/bin/format/elf/elf.c

```diff
@@ -1415,6 +1415,142 @@ ut64 Elf_(rz_bin_elf_get_section_addr_end)(ELFOBJ *bin, const char *section_name
 	RzBinElfSection *section = get_section_by_name(bin, section_name);
 	return section ? section->rva + section->size : UT64_MAX;
 }
+char* Elf_(rz_bin_elf_get_section_flag)(struct rz_bin_elf_section_t * section) {
```

Comment:
Flags should probably be split in int/define values that can be put in or (`|`) instead of just having a string representing them.
If we just use a string, we can't do anything with these flags apart from printing them.

Reply:
However, before doing that one should quickly assess what kind of flags/types are needed for other most common file formats like PE or MACH-O, and see what can be grouped together and what not. For example, I would expect that also other file formats have a concept of "writable section".

Reply:
From my research into the ELF, PE and MACH-O formats I found:

| Format | Has Section Types | Has Section Flags |
| --------- | ------------------------- | ------------------------ |
|   ELF   | ✅                           | ✅                           |
|   PE     | ❌                            | ✅                           |
|  MACH-O | ✅                      | ✅                           |

Looking at the section types it seems that ELF and MACH-O have few types in common. Some of which are:
 - SHT_INIT_ARRAY ≈ S_MOD_INIT_FUNC_POINTERS
 - SHT_FINI_ARRAY ≈ S_MOD_TERM_FUNC_POINTERS
 
 Using an int/define values would require reducing unique section types to common section types. This can be misleading. For example, MACH-O has a section type for 4 byte literals, 8 byte literals, and C string literals. Reducing these all to a DATA type might be good enough but does lose some information.
 
 Would it be worth losing this information for reduced code complexity?
 
 I have not yet looked at common flags.

Reply:
I see LIEF uses int(or enums, if things cannot be `|`) values to store those flags/types/characteristics. See https://lief.quarkslab.com//doc/latest/api/c/pe.html#pe-section-t , https://lief.quarkslab.com//doc/latest/api/c/elf.html#elf-section-t , https://lief.quarkslab.com//doc/latest/api/c/macho.html#macho-section-t

Reply:
We can probably have in each `RzBinPlugin` a `flag2string` method (or something like that) that given the integer/enum value, returns the string associated with it.

Reply:
Good idea for something like `flag2string`. cc @thestr4ng3r